sql/wsrep_schema.cc

```diff
@@ -252,6 +252,7 @@ static int open_table(THD* thd,
 static int open_for_write(THD* thd, const char* table_name, TABLE** table) {
   LEX_CSTRING schema_str= { wsrep_schema_str.c_str(), wsrep_schema_str.length() };
   LEX_CSTRING table_str= { table_name, strlen(table_name) };
+  thd->lex->reset_query_tables_list(TRUE);
```

Comment:
There are different situations in which `open_for_write()` is called: it could be used in a separate thd object, or in some cases it could be used directly by the thd that is executing a transaction. In this latter case, we probably don't want to leave any side effects on the ongoing transaction. Therefore I suggest we make a more conservative fix.
I'd suggest to do something similar to what is done in `wsrep_schema::remove_fragments()`. There we use `reset_n_backup_query_tables_list()`, wihch returns a query tables list which can later restored using `restore_backup_query_tables list`.

Reply:
changed to use backup query table list for the duration of fragment update and fragment meta data update operations. mtr tests are passing, at least